pkg/compactor/block_upload.go

```diff
@@ -262,22 +272,58 @@ func (c *MultitenantCompactor) UploadBlockFile(w http.ResponseWriter, r *http.Re
 	w.WriteHeader(http.StatusOK)
 }
 
-func (c *MultitenantCompactor) completeBlockUpload(ctx context.Context, logger log.Logger, userBkt objstore.Bucket, blockID ulid.ULID, meta metadata.Meta) error {
+func (c *MultitenantCompactor) completeBlockUpload(ctx context.Context, logger log.Logger, userBkt objstore.Bucket, blockID ulid.ULID, meta metadata.Meta) {
 	level.Debug(logger).Log("msg", "completing block upload", "files", len(meta.Thanos.Files))
 
+	// start go routine that updates validation file timestamp once per minute
+	ch := make(chan string)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	ctx, cancel := context.WithCancel(ctx)
+	go c.periodicValidationUpdater(ctx, logger, blockID, userBkt, ch, &wg, cancel)
+
+	if err := c.validateBlock(ctx, blockID, userBkt, meta); err != nil {
+		level.Error(logger).Log("msg", "error while validating block")
```

Comment:
Shouldn't the error itself be included when logging it? I'm also wondering if this error level log message should happen only if the context wasn't canceled, since context cancellation gets logged with warning level in the other goroutine.
```suggestion
		level.Error(logger).Log("msg", "error while validating block", "err", err)
```